src/Raven.Server/Program.cs

```diff
@@ -321,6 +323,7 @@ public static bool RunAsNonInteractive()
                     return; // already done
                 Console.WriteLine("Received graceful exit request...");
                 ShutdownServerMre.Set();
+                ShutdownCompleteMre.WaitOne(TimeSpan.FromSeconds(30)); // on linux we have to keep Unloading event exit last
```

Comment:
Where are you ever setting this?

Reply:
https://github.com/ravendb/ravendb/pull/9151/files#diff-4539864a63aaae59904fb3d5a47f8e22R271